### PR TITLE
Port prefs to use libadwaita

### PR DESCRIPTION
**Pull request by xanf**
This MR restores functionality of prefs window :tada: 

Also it recovers occasionally hidden workspace name in topbar

image

### CODE REVIEW

File: prefsKeybinding.js [hunk 1]
```diff
@@ -774,6 +774,8 @@ const ComboRow = GObject.registerClass(
     }
 
     _updateState() {
+      if (!this._stack) { return; }
```
Comment (xanf) [inline prefsKeybinding.js:777]: `_updateState` is triggered during initialization, when `this._stack` is missing. do nothing during init

File: topbar.js [hunk 2]
```diff
@@ -296,17 +296,9 @@ var WorkspaceMenu = Utils.registerClass(
...
-        try {
-          GLib.spawn_async(
-            null,
-            ["gnome-shell-extension-prefs", "paperwm@hedning:matrix.org"],
-            env,
-            GLib.SpawnFlags.SEARCH_PATH | GLib.SpawnFlags.DO_NOT_REAP_CHILD,
-            null
-          );
-        } catch (e) {}
+        let temp_file = Gio.File.new_for_path(GLib.get_tmp_dir()).get_child('paperwm.workspace')
```
Comment (xanf) [inline topbar.js:299]: Quite ugly, huh? Unfortunately are no more spawned as separate process so previous env trick will not work. Also in `prefs` there is no `global` or `imports.gi.Shell`. File is the best approach I've found

File: topbar.js [hunk 3]
```diff
@@ -589,11 +581,9 @@ function enable() {
       space.label.clutter_text.set_font_description(fontDescription);
     }
   }
-  // TheSola10: drop the modified Activities button since D2P can't
```
Comment (xanf) [inline topbar.js:592]: I believe this one sneaked here and is not intended to be in public

**Comment by Thesola10** [comment]
nice, sorry about the activities button removal, that was a personal change that slipped thru.

Might want to add a pref to toggle it, or a D2P conditional, what do you think?

As for the forced panel transparency, I'm not a fan, there are other exts that can do that (again, like D2P) but it is a feature change and not a port, so idk

**Pull request merged**